dask_ml/feature_extraction/text.py

```diff
@@ -108,3 +112,253 @@ class FeatureHasher(_BaseHasher, sklearn.feature_extraction.text.FeatureHasher):
+
+    def _fit_transform(self, raw_documents, y=None):
+        client = get_client()
+        params = self.get_params()
+        vocabulary = params.pop("vocabulary")
+        del params["use_actors"]
+
+        vocabulary_ = client.submit(Vocabulary, vocabulary=vocabulary, actor=True)
+        vocabulary_actor = vocabulary_.result()
+
+        if self.vocabulary is not None:
+            # Case 1: Just map transform.
+            fixed_vocabulary = True
+            result = raw_documents.map_partitions(
+                _count_vectorizer_transform, vocabulary_actor, params
+            )
+        else:
+            fixed_vocabulary = False
+            # Case 2: learn vocabulary from the data.
+            vocabularies = raw_documents.map_partitions(
+                _build_vocabulary, vocabulary_actor, params
+            )
+            # compute and merge.
+            dask.compute(vocabularies)  # List[Set[str]]
+            # same as before
+            result = raw_documents.map_partitions(
+                _count_vectorizer_transform, vocabulary_actor, params
+            )
+
+        meta = scipy.sparse.eye(0, format="csr", dtype=self.dtype)
+        result = build_array(result, vocabulary_actor.n_features, meta)
+
+        self.vocabulary_actor_ = vocabulary_actor
+        self.vocabulary_ = vocabulary_actor.finalize().result()
+        self.fixed_vocabulary_ = fixed_vocabulary
+
+        return result
+
+    def transform(self, raw_documents):
+        if self.use_actors:
+            return self._transform(raw_documents)
+        else:
+            return self._transform_no_actor(raw_documents)
+
+    def _transform(self, raw_documents):
+        params = self.get_params()
+        del params["vocabulary"]
+        del params["use_actors"]
+
+        check_is_fitted(self, "vocabulary_")
+        result = raw_documents.map_partitions(
+            _count_vectorizer_transform, self.vocabulary_actor_, params
+        )
+        meta = scipy.sparse.eye(0, format="csr", dtype=self.dtype)
+        return build_array(result, self.vocabulary_actor_.n_features, meta)
+
+    def _fit_transform_no_actor(self, raw_documents, y=None):
+        # Just bag for now.
+        # Two cases:
+        # 1. vocabulary provided, easy.
+        # 2. vocabulary learned, harder.
+        params = self.get_params()
+        vocabulary = params.pop("vocabulary")
+        del params["use_actors"]
+
+        if self.vocabulary is not None:
+            # Case 1: Just map transform.
+            fixed_vocabulary = True
+            result = raw_documents.map_partitions(
+                _count_vectorizer_transform_no_actor, dask.delayed(vocabulary), params
+            )
+        else:
+            fixed_vocabulary = False
+            # Case 2: learn vocabulary from the data.
+            vocabularies = raw_documents.map_partitions(
+                _build_vocabulary_no_actor, params
+            )
```

Comment:
For tree reductions I would expect that `apply_concat_apply` probably has a method that lets you supply functions on how to combine the results of intermediate aggregated chunks.  

Alternatively, this thing might already be a `bag`, in which case I would hope that the bag.unique method would work ok.

Reply:
Ah, Bag.distinct